nautobot/docs/development/core/index.md

```diff
@@ -69,6 +69,30 @@ Nautobot adheres to the Semantic Versioning ("SemVer") strategy, which gives us
 
 For more information, please visit [SemVer.org](https://semver.org/).
 
+#### Public Interface
+
+The guiding principals behind Nautobot's SemVer strategy is scoped to adhering to the prescribed public interface. This includes both the application operators and application developer. Anything not explicitly covered here is presumed to not be covered under SemVer.
+
+* The REST API interface within a Version, will not change, but may add to
+    * URL structures
+    * Expected request and response formats
+    * Behavior of each endpoint
+    * Status and error codes
+    * The HTTP verbs
+* Graphql data structures
+* URL structures for primary navigation and feature access
+* Any Django model and their relationships for models provided by Nautobot
```

Comment:
I don't have specific verbiage to suggest as an alternative, but this feels a bit vague to me. We absolutely might introduce new fields or new behaviors to Django models.

Reply:
i added an info statement below, does that help?